docs/adr/0000-database-upgrade-testing.md

```diff
@@ -0,0 +1,60 @@
+# Database Upgrade Testing
+
+* Status: accepted
+* Deciders: [Sander, Arnaud, Wouter]
+* Date: 12/01/2023
+
+## Context and Problem Statement
+
+When making updates to the database schema, it is important to test this.
+
+## Decision Drivers
+
+* We want to test that database update code correctly transforms data produced by older code to data that can be handled by the current code
+* But within the code base, this older code no longer exists
+
+## Considered Options
+
+1. Write data corresponding to the old database schema directly in the testcase to the DB.
+2. Generate test data using old code and persist it.
+3. Dynamically check out old code to run the test
+
+## Decision Outcome
+
+We chose option 2 to test upgrades by
```

Comment:
As we saw in @Hugo-Inmanta 's PR there are two flavors of this option:
a. we always do it like this
b. we are flexible and allow prepopulation in the test case iff it doesn't use any code from `inmanta`.  Directly accessing the database is always safe, since we're operating on a dump that reflects the old data and schema.

Option b could serve to limit pollution of the dump tool for very specific edge cases, and it might simplify the test by making them more directed (they put something in and make sure it comes out as expected, focusing only on that representative subset).
I would consider this to be the best of both worlds.

Reply:
added

Reply:
I don't see it. Forgot to push?

Reply:
:facepalm: